auto_backup/model/backup_scheduler.py

```diff
@@ -0,0 +1,376 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#     OpenERP, Open Source Management Solution
+#     Copyright (C) 2004-2009 Tiny SPRL (<http://tiny.be>). All Rights Reserved
+#     Copyright 2015 Agile Business Group <http://www.agilebg.com>
+#
+#     This program is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU General Public License as published by
+#     the Free Software Foundation, either version 3 of the License, or
+#     (at your option) any later version.
+#
+#     This program is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#
+#     You should have received a copy of the GNU General Public License
+#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import socket
+import os
+import time
+import datetime
+import re
+from openerp import models, fields, api, _
+from openerp.exceptions import except_orm, Warning as UserError
+from openerp import tools
+from openerp.service import db
+import logging
+_logger = logging.getLogger(__name__)
+try:
+    import pysftp
+except ImportError:
+    _logger.debug('Can not import pysftp')
+
+
+def execute(connector, method, *args):
+    res = False
+    try:
+        res = getattr(connector, method)(*args)
+    except socket.error as e:
+        raise e
+    return res
+
+
+class DbBackup(models.Model):
+    _name = 'db.backup'
+
+    @api.model
+    def _get_db_name(self):
+        return self.env.cr.dbname
+
+    name = fields.Char(
+        string='Database', size=100, required=True,
+        default=_get_db_name,
+        help='Database you want to schedule backups for'
+    )
+
+    bkp_dir = fields.Char(
+        string='Backup Directory', size=100,
+        default='/odoo/backups',
```

Comment:
I'd prefer a better default that ensures more directory writeability, like `~/backups`.